src/saveload/saveload.cpp

```diff
@@ -2814,7 +2814,7 @@ static void SaveFileError()
  * We have written the whole game into memory, _memory_savegame, now find
  * and appropriate compressor and start writing to file.
  */
-static SaveOrLoadResult SaveFileToDisk(bool threaded)
+static SaveOrLoadResult SaveFileToDisk(bool threaded, const std::string &filename)
```

Comment:
It feels very confusing to me that filename is only used to display the debug message, it's not actually "used" for generating the actual file. Not sure if there's anything that can be done about that without a much more significant refactor though. Especially when there's a default parameter on `DoSave` that's just empty.

Also, missing doxygen